src/api/middlewares/adminCheck.js

```diff
@@ -0,0 +1,32 @@
+import { StatusCodes } from "http-status-codes";
+import ERoles from "../enums/ERole";
+import { decodeJwtToken } from "../utils/jwt";
+
+const adminCheck = async (req, res, next) => {
+  try {
+    const authHeader = req.headers.authorization;
+    if (!authHeader) {
+      return res
+        .status(401)
+        .json({ message: "Authorization header not found" });
+    }
+
+    const token = authHeader.replace("Bearer ", "");
+    const decoded = await decodeJwtToken(token);
+    const isAdmin = decoded.roles.includes(ERoles.ADMINISTRATOR);
+    if (!isAdmin) {
+      return res
+        .status(StatusCodes.FORBIDDEN)
+        .json({ message: "You are not authorized to access this resource" });
+    }
+
+    // If the user is an admin, pass control to the next middleware
+    next();
+  } catch (err) {
+    return res
+      .status(StatusCodes.UNAUTHORIZED)
+      .json({ message: "Invalid or expired token", error: err.message });
+  }
+};
```

Comment:
refactor this into two middleware function
- check user is logged in using authorization and get the user using id from the decoded user and send to the next middleware
- add a second function to restrict by role
    . function receives many roles and checks if logged user role include in 
   refer to this
     ``export const restrictTo = (...roles)=>{
    return (req, res, next) => {
        const role = req.user.role.title
        if (!roles.includes(role)) {
          return res.status(403).json({
            message: "You are not authorized to perform this action",
          });
        }
        next();
      };
}